src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -813,7 +814,11 @@ Status ServiceBasedNodeInfoAccessor::AsyncGetInternalConfig(
                    std::unordered_map<std::string, std::string>(
                        reply.config().config().begin(), reply.config().config().end()));
         } else {
-          RAY_LOG(ERROR) << "Failed to get internal config: " << status.message();
+          if (!status.ok()) {
+            RAY_LOG(ERROR) << "Failed to get internal config: " << status.message();
+          } else {
+            RAY_LOG(DEBUG) << "No internal config was stored.";
+          }
```

Comment:
Why don't we just do 
```
else if (!status.ok()) {
  RAY_LOG(ERROR) << "something"
}
```

Then we don't even need the last else.

Also, what's the reason why we don't fail when we fail to get internal config?

Reply:
Because it is an empty optional object instead of an empty optional object accompanied with an actual error message.